# lncRNA-analysis-and-preprocessing

# ANALYZING POSSIBLY MISANNOTATED CODING LncRNAs DISCOVERED THROUGH DEEP LEARNING MODELS

## Abstract

Long noncoding RNAs (lncRNAs) are the largest class of noncoding RNAs (ncRNAs). Recent experiments have identified some translatable small open reading frames (sORFs) within these transcripts. Mass spectrometry and ribosome profiling (Ribo-Seq) are the current experimental methods used to identify misannotated lncRNAs but both associated with many false positives. In their work, Nabi et al used deep learning models to classify RNA transcripts. They suspected that false negatives with a high coding probability could be misannotated. Our study aims to find other evidence to support this hypothesis. From our analyses: a good number of the suspected misannotated candidates contained exons, had blast hits, and several disease-associated variants.

Keywords: lncRNA, disease associated variants, conserved region, mis-annotated RNA.

## 1 Introduction

Transcripts that do not have the protein-coding capacity in the human genome are called non-coding RNA (ncRNA). These non-coding RNAs are of two main categories according to their length as small non-coding RNAs (sncRNA) and long non-coding RNAs (lncRNA). Transcripts that are more than 200 nucleotides in length in a non-coding RNA sequence are called lncRNAs. A small portion of which are functionally characterized are known to play vital roles in translation regulation, transcription, chromatin modification and mRNA stability (Batista & Chang, 2013; Rinn & Chang, 2012; Ulitsky & Bartel, 2013). Recent studies have shown that some lncRNAs contain small open reading frames (sORFs) translated into functional micro peptides with a median length of 23 amino acids (Ingolia et al., 2009; Long Non-Coding RNAs as a Source of New Peptides, 2014; Lu et al., 2019) and perform vital functions across species including bacteria, flies and humans (Batista & Chang, 2013; Makarewich et al., 2018; Nelson et al., 2016; Matsumoto, 2017). Therefore, the identification of misinterpreted lncRNAs is important for the functional characterization of these transcripts and the proteome. Ribosome profiling (Ribo-Seq) and mass spectrometry (MS) are the main experimental methods used for identifying misannotated lncRNA (Slavoff et al., 2012; Ingolia, 2014). However, MS is not very accurate and Ribo-Seq contains many false positives. For this reason, many computational algorithms have been developed to distinguish false positives from true positives including FLOSS (Ingolia, 2014)  and ORFscore (Bazzini et al., 2014). However, each of these methods has its shortcomings. In recent work, deep learning models were trained to classify between lncRNA and cRNA and to use the deep learning training dynamics in identifying possibly misannotated lncRNAs (Nabi, 2021). Figure 1 below shows the proportion of RNAs that were tagged as misannotated (in black). We aim to find further supporting evidence to show that these putative mislabelled RNAs detected by deep learning models may be indeed misannotated.

![Ekran görüntüsü 2022-09-28 202220](https://user-images.githubusercontent.com/71135791/192848235-009e4efe-94a1-46ff-936f-1f35677a92bb.jpg)

## Figure 1: The t-SNE clusters obtained from the features generated in the nucleotide sequences are shown (Lu et al., 2019). Putative misinterpreted lncRNAs and coding RNAs together are in a single cluster (Nabi, 2021).

lncRNAs are known for their non-coding capacity and non-conservability (do not contain conserved regions). Therefore, if any evidence is found that does not comply with these properties, such evidence may support the hypothesis that these lncRNAs are misannotated. For Example, it would be interesting to search for disease-associated variants, exons, blast hits and conserved regions with these transcripts. A blast hit is when a similar protein structure is associated with a particular sequence (DNA, RNA or protein) during a blast search. Normally, lncRNAs should not be associated with any protein structure since they do code for proteins. Also, it will be interesting to find exons with a non-coding transcript as exons are protein-coding sequences. Moreso, The presence of a disease-associated variant, especially a benign variant may indicate that these transcripts code for malfunctional protein that must be seriously considered for medical practices and research. Finally, it will be interesting to examine the polypeptide structures of the ORFs within transcript candidates that have several supporting pieces of evidence. Thus our research is divided into the following steps: finding genomic coordinates for each transcript using its Ensembl transcript ID from the Ensembl database (Cunningham et al., 2021), finding all disease-associated variant positions with the transcript coordinates and their clinical significance using the Clinvar database (Landrum et al., 2017), and finding all possible exons from with each transcript using the Ensembl database. Blast search results are obtained from a previous study (Nabi, 2021), and the examination of the ORFs for strong candidate transcripts is considered a future study.

## 2 Methods

The dataset used for this research is obtained by selecting lncRNAs that are persistently being predicted by deep learning models as coding RNA with a high probability (Nabi, 2021). This dataset is in tabular form with many columns. However, the columns of interest are the query columns which contain transcript IDs (e.g ENST00000621052) and the sequence column which contains the blast ID (e.g O00370) of a protein that is most similar to the transcript sequence. The Clinvar database contains disease-associated variant positions and their clinical significance. The Ensembl database on the other hand can be used to retrieve transcript coordinates and exons given their Ensembl transcript ID. Figure 2 below shows how different elements were extracted from different data sources and each step will be explained in details in the following subsections.

![Ekran görüntüsü 2022-09-28 202400](https://user-images.githubusercontent.com/71135791/192848548-bc4d7b4e-f679-46d4-acb6-c22df0ff6f95.jpg)

## Figure 2. Flow chart of data extraction from different data sources

## 2.1 Fetching Transcript Coordinates

To find disease-associated variants from the Clinvar database, one needs to have target genomic positions. The Ensembl database contains several transcript information for many different species. Pyensembl (PyEnsembl Is a Python Interface to Ensembl Reference Genome Metadata such as Exons and Transcripts | CuratedPython, 2021), a python API for Ensembl is used to download the Ensembl database on any local PC. Since we are working with human RNA transcripts, the latest Ensembl release (release 106) for homo sapiens is downloaded. This release contains data for assembly GRCh38. If multiple ensemble releases have been downloaded on the same local machine, it is necessary to specify the release version during computation to avoid crashes between two or more different genome assemblies. Using the EnsemblRelease module from Pyensembl, given a transcript ID, the chromosome, strand, start position, and end positions are extracted. Table 1 below shows a few rows and important columns of the results obtained from the Ensembl database.

![Ekran görüntüsü 2022-09-28 202516](https://user-images.githubusercontent.com/71135791/192849060-9b043395-1223-4cab-a38c-9063d6f349b6.jpg)

## Table 1. Sample results obtained from ensembl database using the Pyensembl Transcripts module.

## 2.2 Fetching Disease Associated Variants

The ClinVar database is a specialized database that can be used to find the positions of disease-associated variants in RNA and DNA sequences. The ClinVar database contains information on the position of disease-bearing genes of various species. The Homo sapiens dataset was downloaded to the local workspace. Two files were downloaded; a vcf file (https://ftp.ncbi.nlm.nih.gov/pub/clinvar/vcf_GRCh38/) that contains about 1.4 million unique disease variant positions and a Clinvar table in CSV format obtained through a variant search in the Clinvar database that contains the clinical significance of disease variant positions. Both files were analyzed and merged. To determine whether there is a disease variant in the data set, the Ensembl python API mentioned under the heading 2.1 is used to obtain transcript coordinates (Start position, end position and chromosome number). If there is a disease variant in any nucleotide between the start and end position, this nucleotide position is recorded. It would be time-consuming and computationally expensive to check for each position within the transcript coordinate if there is a disease variant or not. To solve this computational redundancy, a chromosome variant map is created in which each unique chromosome is associated with all the disease variant positions in it. Therefore, given a transcript coordinate, the chromosome number is used to find all possible disease variants on the given chromosome, and for each variant, it is checked whether it falls within the start and end positions of the given transcript. When these operations are applied to all transcript IDs, the disease-associated variants in the data set are fetched and assigned to their respective transcripts and the results are stored in a dictionary for further retrieval. Table 2 below shows sample results of disease-associated variants found in some transcripts.

![Ekran görüntüsü 2022-09-28 202531](https://user-images.githubusercontent.com/71135791/192849300-bd23c439-ba80-4f43-a7ff-19583a7b1a25.jpg)

## Table 2.  Sample results from ClinVar database

## 2.3 Fetching Exons and Blast Hits 

The EnsemblRelease module also has a special method for retrieving exons associated with any transcript given its ID. Using this module, information such as all exon ids associated with the given transcript and exon coordinates (chromosome, start and end positions) of each transcript were retrieved. To validate the results, the exon coordinates were checked for being within the transcript coordinate and all exons passed the test. The dataset used in this research already contained Blast results and so transcripts that were associated with a blast Hit were annotated accordingly. 

Results obtained from all three sections above are combined and analyzed. Scatter plots and bar charts were plotted to show the distribution of transcripts within the genome and which transcripts meet the research criteria. The results are further discussed in the next section.

## 3 Discussion and Conclusion 

![Ekran görüntüsü 2022-09-28 202550](https://user-images.githubusercontent.com/71135791/192849465-6e25ee9c-c7bc-4467-b023-cd32d5af78a8.jpg)

## Figure 3. Scatter plot showing transcripts with or without disease-associated variants

![Ekran görüntüsü 2022-09-28 202606](https://user-images.githubusercontent.com/71135791/192849493-ab93eba6-7fef-4722-a08e-7dbde0c58d36.jpg)

## Figure 4. Plot showing the presence of different kinds of disease  variants within transcripts

![Ekran görüntüsü 2022-09-28 202622](https://user-images.githubusercontent.com/71135791/192849595-da52bed1-b5f5-4bb1-b7d1-f91d2193b04e.jpg)

## Figure 5. Plot showing the amount of transcripts that satisfy all each criteria (presence of disease-associated variants, exons and blast hits)

Figures 3, 4 and 5 show the results of our analysis. In Figure 3, the focus was to find transcripts with disease-associated variants and such transcripts are indicated in orange while transcripts with no disease-associated variants are indicated in sky blue colour. Also, the number of disease-associated variants is used to scale the dot representation of each transcript. As a result, it may be noticed that some dots are slightly larger than others (especially orange dots) indicating that such transcripts contain more disease-associated variants. Through manual observation, some transcripts were found to contain as many as 300 disease-associated variants (e.g ENST00000627564(5055), ENST00000501371(537), ENST00000420707(395), etc). In Figure 4 on the other hand, only transcripts with disease-associated variants are plotted. The different variant types are specified in different colours. That is, transcripts that contain; benign variants (BV) in blue, pathogenic variants (PV) in green, both benign and pathogenic variants in orange, likely benign or pathogenic variants in violet and other types of variants in red. It should be noted that a point coloured orange, for example, does necessarily entail that the transcript contains only benign and pathogenic variants. Some transcripts may even contain all sorts of variants in their numbers but were represented graphically based on the chronological occurrence of PV and BV, BV, PV, likely PV or BV and Others (to represent any other variants i.e not provided, uncertain, etc). Whatever the case, 192 transcripts have been associated with disease variants, most of which are benign variants (a total of 169).   Finally, Figure 5 shows the results of the different lines of support (presence of exons, blast hits and disease-associated variants). Surprisingly all 1233 transcripts contain one or more exons of significant length and were all associated with a similar protein due to the presence of a blast in each case. And these transcripts are almost evenly distributed among the different types of chromosomes.

The results of our findings may be a clear indication that these transcript sequences have been misannotated. However, we still intend to find more supporting evidence such as finding conserved regions within the transcripts. It is known that lncRNAs do not contain conserved regions. Therefore, if we can find conserved regions within these transcripts, it could be supporting evidence that these transcripts are misannotated. Also, 169 transcripts showed many lines of support ( contained benign variants, contained exons and were associated with a blast hit). Performing an Alpha-Fold on the ORFs of these transcripts to examine their structures will help validate this methodology.  Finally, our research has convinced us that deep learning models can be used for identifying possibly misannotated lncRNAs.

## References 

Batista, Pedro J., & Chang, Howard Y. (2013). Long Noncoding RNAs: Cellular Address Codes in Development and Disease. Cell, 152(6), 1298–1307. https://doi.org/10.1016/j.cell.2013.02.012
Ingolia, N. T., Ghaemmaghami, S., Newman, J. R. S., & Weissman, J. S. (2009). Genome-Wide Analysis in Vivo of Translation with Nucleotide Resolution Using Ribosome Profiling. Science, 324(5924), 218–223. https://doi.org/10.1126/science.1168978
Ruiz-Orera, J., Messeguer, X., Subirana, J. A., & Alba, M. M. (2014). Long non-coding RNAs as a source of new peptides. ELife, 3. https://doi.org/10.7554/elife.03523
Lu, S., Zhang, J., Lian, X., Sun, L., Meng, K., Chen, Y., … He, Q.-Y. (2019). A hidden human proteome encoded by “non-coding” genes. Nucleic Acids Research, 47(15), 8111–8125. https://doi.org/10.1093/nar/gkz646
Nabi, A. (2021). Discovering coding LNCRNAS using deep learning training dynamics. Research.sabanciuniv.edu. https://research.sabanciuniv.edu/id/eprint/42505/
Rinn, J. L., & Chang, H. Y. (2012). Genome Regulation by Long Noncoding RNAs. Annual Review of Biochemistry, 81(1), 145–166. https://doi.org/10.1146/annurev-biochem-051410-092902
Ulitsky, I., & Bartel, David P. (2013). lincRNAs: Genomics, Evolution, and Mechanisms. Cell, 154(1), 26–46. https://doi.org/10.1016/j.cell.2013.06.020
Makarewich, C. A., Baskin, K. K., Munir, A. Z., Bezprozvannaya, S., Sharma, G., Khemtong, C., Shah, A. M., McAnally, J. R., Malloy, C. R., Szweda, L. I., Bassel-Duby, R., & Olson, E. N. (2018). MOXI Is a Mitochondrial Micropeptide That Enhances Fatty Acid β-Oxidation. Cell Reports, 23(13), 3701–3709. https://doi.org/10.1016/j.celrep.2018.05.058
Nelson, B. R., Makarewich, C. A., Anderson, D. M., Winders, B. R., Troupes, C. D., Wu, F., Reese, A. L., McAnally, J. R., Chen, X., Kavalali, E. T., Cannon, S. C., Houser, S. R., Bassel-Duby, R., & Olson, E. N. (2016). A peptide encoded by a transcript annotated as long noncoding RNA enhances SERCA activity in muscle. Science, 351(6270), 271–275. https://doi.org/10.1126/science.aad4076
Matsumoto, A., Pasut, A., Matsumoto, M., Yamashita, R., Fung, J., Monteleone, E., Saghatelian, A., Nakayama, K. I., Clohessy, J. G., & Pandolfi, P. P. (2016). mTORC1 and muscle regeneration are regulated by the LINC00961-encoded SPAR polypeptide. Nature, 541(7636), 228–232. https://doi.org/10.1038/nature21034
Slavoff, S. A., Mitchell, A. J., Schwaid, A. G., Cabili, M. N., Ma, J., Levin, J. Z., Karger, A. D., Budnik, B. A., Rinn, J. L., & Saghatelian, A. (2012). Peptidomic discovery of short open reading frame–encoded peptides in human cells. Nature Chemical Biology, 9(1), 59–64. https://doi.org/10.1038/nchembio.1120
Bazzini, A. A., Johnstone, T. G., Christiano, R., Mackowiak, S. D., Obermayer, B., Fleming, E. S., Vejnar, C. E., Lee, M. T., Rajewsky, N., Walther, T. C., & Giraldez, A. J. (2014). Identification of small ORFs in vertebrates using ribosome footprinting and evolutionary conservation. The EMBO Journal, 33(9), 981–993. https://doi.org/10.1002/embj.201488411
Cunningham, F., Allen, J. E., Allen, J., Alvarez-Jarreta, J., Amode, M., Armean, I., Austine-Orimoloye, O., Azov, A., Barnes, I., Bennett, R., Berry, A., Bhai, J., Bignell, A., Billis, K., Boddu, S., Brooks, L., Charkhchi, M., Cummins, C., Da Rin Fioretto, L., & Davidson, C. (2021). Ensembl 2022. Nucleic Acids Research, 50(D1), D988–D995. https://doi.org/10.1093/nar/gkab1049
Landrum, M. J., Lee, J. M., Benson, M., Brown, G. R., Chao, C., Chitipiralla, S., Gu, B., Hart, J., Hoffman, D., Jang, W., Karapetyan, K., Katz, K., Liu, C., Maddipatla, Z., Malheiro, A., McDaniel, K., Ovetsky, M., Riley, G., Zhou, G., & Holmes, J. (2017). ClinVar: improving access to variant interpretations and supporting evidence. Nucleic Acids Research, 46(D1), D1062–D1067. https://doi.org/10.1093/nar/gkx1153
PyEnsembl is a Python interface to Ensembl reference genome metadata such as exons and transcripts | CuratedPython. (2021, October 15). Curatedpython.com. https://curatedpython.com/p/pyensembl-is-openvax-pyensembl/index.html



