{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "2edc9609",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "9699183a",
   "metadata": {},
   "outputs": [],
   "source": [
    "Data=pd.read_csv('blast_dyn_ribo_merged_candidates - blast_dyn_ribo_merged_candidates.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "75a09fe2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>query</th>\n",
       "      <th>length</th>\n",
       "      <th>sequence</th>\n",
       "      <th>score</th>\n",
       "      <th>expect</th>\n",
       "      <th>method</th>\n",
       "      <th>identities</th>\n",
       "      <th>positives</th>\n",
       "      <th>gaps</th>\n",
       "      <th>...</th>\n",
       "      <th>Annotation</th>\n",
       "      <th>Biotype</th>\n",
       "      <th>Floss-Classification</th>\n",
       "      <th>FLOSS score</th>\n",
       "      <th>Orfscore</th>\n",
       "      <th>PhyloP</th>\n",
       "      <th>cutoff p-value</th>\n",
       "      <th>PhastCon</th>\n",
       "      <th>Ensembl transcript ID</th>\n",
       "      <th>PeptideShaker confidence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>ENST00000621052</td>\n",
       "      <td>1275</td>\n",
       "      <td>O00370</td>\n",
       "      <td>140.0</td>\n",
       "      <td>4.000000e-32</td>\n",
       "      <td>Compositional matrix adjust.</td>\n",
       "      <td>66/84 (79%)</td>\n",
       "      <td>73/84 (87%)</td>\n",
       "      <td>0/84 (0%)</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>ENST00000549804</td>\n",
       "      <td>1032</td>\n",
       "      <td>B0Y4D6</td>\n",
       "      <td>37.0</td>\n",
       "      <td>3.200000e+00</td>\n",
       "      <td>Compositional matrix adjust.</td>\n",
       "      <td>26/73 (36%)</td>\n",
       "      <td>39/73 (53%)</td>\n",
       "      <td>4/73 (5%)</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>ENST00000415801</td>\n",
       "      <td>159</td>\n",
       "      <td>Q96CB5</td>\n",
       "      <td>64.3</td>\n",
       "      <td>2.000000e-10</td>\n",
       "      <td>Compositional matrix adjust.</td>\n",
       "      <td>31/44 (70%)</td>\n",
       "      <td>36/44 (82%)</td>\n",
       "      <td>0/44 (0%)</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "      <td>ENST00000609590</td>\n",
       "      <td>367</td>\n",
       "      <td>O54689</td>\n",
       "      <td>557.0</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>Compositional matrix adjust.</td>\n",
       "      <td>276/370 (75%)</td>\n",
       "      <td>320/370 (86%)</td>\n",
       "      <td>4/370 (1%)</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>ENST00000523176</td>\n",
       "      <td>683</td>\n",
       "      <td>O94929</td>\n",
       "      <td>69.3</td>\n",
       "      <td>2.000000e-10</td>\n",
       "      <td>Compositional matrix adjust.</td>\n",
       "      <td>30/30 (100%)</td>\n",
       "      <td>30/30 (100%)</td>\n",
       "      <td>0/30 (0%)</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1318</th>\n",
       "      <td>1318</td>\n",
       "      <td>ENST00000588290</td>\n",
       "      <td>531</td>\n",
       "      <td>Q9NV72</td>\n",
       "      <td>38.9</td>\n",
       "      <td>1.200000e-02</td>\n",
       "      <td>Composition-based stats.</td>\n",
       "      <td>19/30 (63%)</td>\n",
       "      <td>20/30 (67%)</td>\n",
       "      <td>0/30 (0%)</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1319</th>\n",
       "      <td>1319</td>\n",
       "      <td>ENST00000521197</td>\n",
       "      <td>401</td>\n",
       "      <td>Q5IS58</td>\n",
       "      <td>35.8</td>\n",
       "      <td>2.700000e-01</td>\n",
       "      <td>Compositional matrix adjust.</td>\n",
       "      <td>22/61 (36%)</td>\n",
       "      <td>33/61 (54%)</td>\n",
       "      <td>2/61 (3%)</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1320</th>\n",
       "      <td>1320</td>\n",
       "      <td>ENST00000641534</td>\n",
       "      <td>464</td>\n",
       "      <td>P04198</td>\n",
       "      <td>71.2</td>\n",
       "      <td>3.000000e-14</td>\n",
       "      <td>Compositional matrix adjust.</td>\n",
       "      <td>32/32 (100%)</td>\n",
       "      <td>32/32 (100%)</td>\n",
       "      <td>0/32 (0%)</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1321</th>\n",
       "      <td>1321</td>\n",
       "      <td>ENST00000528233</td>\n",
       "      <td>230</td>\n",
       "      <td>Q96NF6</td>\n",
       "      <td>66.6</td>\n",
       "      <td>3.000000e-12</td>\n",
       "      <td>Compositional matrix adjust.</td>\n",
       "      <td>35/55 (64%)</td>\n",
       "      <td>37/55 (67%)</td>\n",
       "      <td>2/55 (4%)</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1322</th>\n",
       "      <td>1322</td>\n",
       "      <td>ENST00000599924</td>\n",
       "      <td>174</td>\n",
       "      <td>Q8N2A0</td>\n",
       "      <td>59.3</td>\n",
       "      <td>8.000000e-11</td>\n",
       "      <td>Compositional matrix adjust.</td>\n",
       "      <td>31/43 (72%)</td>\n",
       "      <td>34/43 (79%)</td>\n",
       "      <td>2/43 (5%)</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1323 rows × 33 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Unnamed: 0            query  length sequence  score        expect  \\\n",
       "0              0  ENST00000621052    1275   O00370  140.0  4.000000e-32   \n",
       "1              1  ENST00000549804    1032   B0Y4D6   37.0  3.200000e+00   \n",
       "2              2  ENST00000415801     159   Q96CB5   64.3  2.000000e-10   \n",
       "3              3  ENST00000609590     367   O54689  557.0  0.000000e+00   \n",
       "4              4  ENST00000523176     683   O94929   69.3  2.000000e-10   \n",
       "...          ...              ...     ...      ...    ...           ...   \n",
       "1318        1318  ENST00000588290     531   Q9NV72   38.9  1.200000e-02   \n",
       "1319        1319  ENST00000521197     401   Q5IS58   35.8  2.700000e-01   \n",
       "1320        1320  ENST00000641534     464   P04198   71.2  3.000000e-14   \n",
       "1321        1321  ENST00000528233     230   Q96NF6   66.6  3.000000e-12   \n",
       "1322        1322  ENST00000599924     174   Q8N2A0   59.3  8.000000e-11   \n",
       "\n",
       "                            method     identities      positives        gaps  \\\n",
       "0     Compositional matrix adjust.    66/84 (79%)    73/84 (87%)   0/84 (0%)   \n",
       "1     Compositional matrix adjust.    26/73 (36%)    39/73 (53%)   4/73 (5%)   \n",
       "2     Compositional matrix adjust.    31/44 (70%)    36/44 (82%)   0/44 (0%)   \n",
       "3     Compositional matrix adjust.  276/370 (75%)  320/370 (86%)  4/370 (1%)   \n",
       "4     Compositional matrix adjust.   30/30 (100%)   30/30 (100%)   0/30 (0%)   \n",
       "...                            ...            ...            ...         ...   \n",
       "1318      Composition-based stats.    19/30 (63%)    20/30 (67%)   0/30 (0%)   \n",
       "1319  Compositional matrix adjust.    22/61 (36%)    33/61 (54%)   2/61 (3%)   \n",
       "1320  Compositional matrix adjust.   32/32 (100%)   32/32 (100%)   0/32 (0%)   \n",
       "1321  Compositional matrix adjust.    35/55 (64%)    37/55 (67%)   2/55 (4%)   \n",
       "1322  Compositional matrix adjust.    31/43 (72%)    34/43 (79%)   2/43 (5%)   \n",
       "\n",
       "      ...  Annotation Biotype Floss-Classification  FLOSS score  Orfscore  \\\n",
       "0     ...         NaN     NaN                  NaN          NaN       NaN   \n",
       "1     ...         NaN     NaN                  NaN          NaN       NaN   \n",
       "2     ...         NaN     NaN                  NaN          NaN       NaN   \n",
       "3     ...         NaN     NaN                  NaN          NaN       NaN   \n",
       "4     ...         NaN     NaN                  NaN          NaN       NaN   \n",
       "...   ...         ...     ...                  ...          ...       ...   \n",
       "1318  ...         NaN     NaN                  NaN          NaN       NaN   \n",
       "1319  ...         NaN     NaN                  NaN          NaN       NaN   \n",
       "1320  ...         NaN     NaN                  NaN          NaN       NaN   \n",
       "1321  ...         NaN     NaN                  NaN          NaN       NaN   \n",
       "1322  ...         NaN     NaN                  NaN          NaN       NaN   \n",
       "\n",
       "      PhyloP  cutoff p-value  PhastCon  Ensembl transcript ID  \\\n",
       "0        NaN             NaN       NaN                    NaN   \n",
       "1        NaN             NaN       NaN                    NaN   \n",
       "2        NaN             NaN       NaN                    NaN   \n",
       "3        NaN             NaN       NaN                    NaN   \n",
       "4        NaN             NaN       NaN                    NaN   \n",
       "...      ...             ...       ...                    ...   \n",
       "1318     NaN             NaN       NaN                    NaN   \n",
       "1319     NaN             NaN       NaN                    NaN   \n",
       "1320     NaN             NaN       NaN                    NaN   \n",
       "1321     NaN             NaN       NaN                    NaN   \n",
       "1322     NaN             NaN       NaN                    NaN   \n",
       "\n",
       "     PeptideShaker confidence  \n",
       "0                         NaN  \n",
       "1                         NaN  \n",
       "2                         NaN  \n",
       "3                         NaN  \n",
       "4                         NaN  \n",
       "...                       ...  \n",
       "1318                      NaN  \n",
       "1319                      NaN  \n",
       "1320                      NaN  \n",
       "1321                      NaN  \n",
       "1322                      NaN  \n",
       "\n",
       "[1323 rows x 33 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from pyensembl import EnsemblRelease\n",
    "data = EnsemblRelease(release=106)\n",
    "Data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "d9e03a70",
   "metadata": {},
   "outputs": [],
   "source": [
    "#obj = data.transcript_by_id(transcript_id = \"ENST00000369985\")\n",
    "transcriptOf=[]\n",
    "error_list=[]\n",
    "for i in Data.index:\n",
    "    try:\n",
    "        transcriptOf.append(data.transcript_by_id(transcript_id = str(Data['query'][i])))\n",
    "    except:\n",
    "        continue\n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "82a57e3b",
   "metadata": {},
   "outputs": [],
   "source": [
    "data_list=[]\n",
    "for i in range(len(transcriptOf)):\n",
    "    data_list.append([transcriptOf[i].transcript_id,transcriptOf[i].transcript_name,transcriptOf[i].gene_id,transcriptOf[i].biotype,\n",
    "                     transcriptOf[i].contig,transcriptOf[i].start,transcriptOf[i].end,transcriptOf[i].strand,transcriptOf[i].genome,\n",
    "                     ])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "7a873653",
   "metadata": {},
   "outputs": [],
   "source": [
    "data_list_columns=['transcript_id','transcript_name','gene_id','biotype','contig','start','end','strand','genome']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "fdd1ba05",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>transcript_name</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>biotype</th>\n",
       "      <th>contig</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>strand</th>\n",
       "      <th>genome</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENST00000621052</td>\n",
       "      <td>LINC01297-202</td>\n",
       "      <td>ENSG00000274827</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>14</td>\n",
       "      <td>19348216</td>\n",
       "      <td>19384587</td>\n",
       "      <td>-</td>\n",
       "      <td>EnsemblRelease(release=106, species='homo_sapi...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENST00000549804</td>\n",
       "      <td>SNHG14-210</td>\n",
       "      <td>ENSG00000224078</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>15</td>\n",
       "      <td>25050632</td>\n",
       "      <td>25138756</td>\n",
       "      <td>+</td>\n",
       "      <td>EnsemblRelease(release=106, species='homo_sapi...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENST00000415801</td>\n",
       "      <td>GAS1RR-201</td>\n",
       "      <td>ENSG00000226237</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>9</td>\n",
       "      <td>86948699</td>\n",
       "      <td>87002033</td>\n",
       "      <td>+</td>\n",
       "      <td>EnsemblRelease(release=106, species='homo_sapi...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENST00000609590</td>\n",
       "      <td></td>\n",
       "      <td>ENSG00000272980</td>\n",
       "      <td>retained_intron</td>\n",
       "      <td>6</td>\n",
       "      <td>166999405</td>\n",
       "      <td>167139141</td>\n",
       "      <td>+</td>\n",
       "      <td>EnsemblRelease(release=106, species='homo_sapi...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENST00000523176</td>\n",
       "      <td></td>\n",
       "      <td>ENSG00000253406</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>5</td>\n",
       "      <td>149216523</td>\n",
       "      <td>149276805</td>\n",
       "      <td>-</td>\n",
       "      <td>EnsemblRelease(release=106, species='homo_sapi...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1228</th>\n",
       "      <td>ENST00000588290</td>\n",
       "      <td>BSG-AS1-201</td>\n",
       "      <td>ENSG00000267751</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>19</td>\n",
       "      <td>567433</td>\n",
       "      <td>572228</td>\n",
       "      <td>-</td>\n",
       "      <td>EnsemblRelease(release=106, species='homo_sapi...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1229</th>\n",
       "      <td>ENST00000521197</td>\n",
       "      <td>HOXA-AS3-203</td>\n",
       "      <td>ENSG00000254369</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>7</td>\n",
       "      <td>27140364</td>\n",
       "      <td>27152697</td>\n",
       "      <td>+</td>\n",
       "      <td>EnsemblRelease(release=106, species='homo_sapi...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1230</th>\n",
       "      <td>ENST00000641534</td>\n",
       "      <td>MYCNOS-204</td>\n",
       "      <td>ENSG00000233718</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>2</td>\n",
       "      <td>15940185</td>\n",
       "      <td>15942249</td>\n",
       "      <td>-</td>\n",
       "      <td>EnsemblRelease(release=106, species='homo_sapi...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1231</th>\n",
       "      <td>ENST00000528233</td>\n",
       "      <td>RSF1-IT1-201</td>\n",
       "      <td>ENSG00000255409</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>11</td>\n",
       "      <td>77738680</td>\n",
       "      <td>77739568</td>\n",
       "      <td>-</td>\n",
       "      <td>EnsemblRelease(release=106, species='homo_sapi...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1232</th>\n",
       "      <td>ENST00000599924</td>\n",
       "      <td></td>\n",
       "      <td>ENSG00000268746</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>19</td>\n",
       "      <td>47607524</td>\n",
       "      <td>47733098</td>\n",
       "      <td>+</td>\n",
       "      <td>EnsemblRelease(release=106, species='homo_sapi...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1233 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        transcript_id transcript_name          gene_id          biotype  \\\n",
       "0     ENST00000621052   LINC01297-202  ENSG00000274827           lncRNA   \n",
       "1     ENST00000549804      SNHG14-210  ENSG00000224078           lncRNA   \n",
       "2     ENST00000415801      GAS1RR-201  ENSG00000226237           lncRNA   \n",
       "3     ENST00000609590                  ENSG00000272980  retained_intron   \n",
       "4     ENST00000523176                  ENSG00000253406           lncRNA   \n",
       "...               ...             ...              ...              ...   \n",
       "1228  ENST00000588290     BSG-AS1-201  ENSG00000267751           lncRNA   \n",
       "1229  ENST00000521197    HOXA-AS3-203  ENSG00000254369           lncRNA   \n",
       "1230  ENST00000641534      MYCNOS-204  ENSG00000233718           lncRNA   \n",
       "1231  ENST00000528233    RSF1-IT1-201  ENSG00000255409           lncRNA   \n",
       "1232  ENST00000599924                  ENSG00000268746           lncRNA   \n",
       "\n",
       "     contig      start        end strand  \\\n",
       "0        14   19348216   19384587      -   \n",
       "1        15   25050632   25138756      +   \n",
       "2         9   86948699   87002033      +   \n",
       "3         6  166999405  167139141      +   \n",
       "4         5  149216523  149276805      -   \n",
       "...     ...        ...        ...    ...   \n",
       "1228     19     567433     572228      -   \n",
       "1229      7   27140364   27152697      +   \n",
       "1230      2   15940185   15942249      -   \n",
       "1231     11   77738680   77739568      -   \n",
       "1232     19   47607524   47733098      +   \n",
       "\n",
       "                                                 genome  \n",
       "0     EnsemblRelease(release=106, species='homo_sapi...  \n",
       "1     EnsemblRelease(release=106, species='homo_sapi...  \n",
       "2     EnsemblRelease(release=106, species='homo_sapi...  \n",
       "3     EnsemblRelease(release=106, species='homo_sapi...  \n",
       "4     EnsemblRelease(release=106, species='homo_sapi...  \n",
       "...                                                 ...  \n",
       "1228  EnsemblRelease(release=106, species='homo_sapi...  \n",
       "1229  EnsemblRelease(release=106, species='homo_sapi...  \n",
       "1230  EnsemblRelease(release=106, species='homo_sapi...  \n",
       "1231  EnsemblRelease(release=106, species='homo_sapi...  \n",
       "1232  EnsemblRelease(release=106, species='homo_sapi...  \n",
       "\n",
       "[1233 rows x 9 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df=pd.DataFrame(data_list,columns=data_list_columns)\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "6f7848e3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['Unnamed: 0', 'query', 'length', 'sequence', 'score', 'expect',\n",
       "       'method', 'identities', 'positives', 'gaps', 'frame', 'ncRNA_biotype',\n",
       "       'RNA', 'mean_cnn', 'std_cnn', 'mean_lstm', 'std_lstm', 'mean_transf',\n",
       "       'std_transf', 'Cell line', 'Chromosome', 'Sorf length', 'Start codon',\n",
       "       'Annotation', 'Biotype', 'Floss-Classification', 'FLOSS score',\n",
       "       'Orfscore', 'PhyloP', 'cutoff p-value', 'PhastCon',\n",
       "       'Ensembl transcript ID', 'PeptideShaker confidence'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Data.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "f5afc969",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0       0.002096\n",
       "1       0.005277\n",
       "2       0.001986\n",
       "3       0.002971\n",
       "4       0.001884\n",
       "          ...   \n",
       "1318    0.136203\n",
       "1319    0.201907\n",
       "1320    0.292935\n",
       "1321    0.202062\n",
       "1322    0.194720\n",
       "Name: std_cnn, Length: 1323, dtype: float64"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Data['std_cnn']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1d0f2cf8",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
